src/lib.rs

```diff
@@ -203,24 +192,23 @@ pub unsafe extern "C" fn spring_pop_non_blocking(
     queue: *const c_char,
     is_err: *mut bool,
 ) -> *mut SpringRow {
-    *is_err = false;
-
-    let pipeline = &*((*pipeline).0 as *const springql_core::SpringPipeline);
+    let pipe = (*pipeline).pipe();
     let queue = CStr::from_ptr(queue).to_string_lossy().into_owned();
-
-    with_catch(|| springql_core::spring_pop_non_blocking(pipeline, &queue)).map_or_else(
-        |_| {
+    let result = with_catch(|| pipe.pop_non_blocking(&queue));
+    match result {
+        Ok(Some(row)) => {
+            let ptr = SpringRow::new(row);
+            ptr.into_ptr()
```

Comment:
Should set `*is_err = false` in this path.

I recommend to initialize it like https://github.com/SpringQL/SpringQL-client-c/pull/36/files#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759L206